src/graph.jl

```diff
@@ -69,6 +79,17 @@ end
 maximum_degree(g::Graph) = maximum(Base.Fix1(degree, g), vertices(g))
 minimum_degree(g::Graph) = minimum(Base.Fix1(degree, g), vertices(g))
 
+"""
+    transpose(g::Graph)
+
+Return a [`Graph`](@ref) corresponding to the transpose of (the underlying matrix of) `g`.
+"""
+function Base.transpose(g::Graph{loops,T}) where {loops,T}
+    S = SparseMatrixCSC{T,T}(g.m, g.n, g.colptr, g.rowval, g.rowval)
+    Sᵀ = convert(SparseMatrixCSC, transpose(S))  # TODO: use ftranspose! without segfault?
```

Comment:
@amontoison my initial idea was to use [`ftranspose!`](https://docs.julialang.org/en/v1.11.0-rc4/stdlib/SparseArrays/#SparseArrays.ftranspose!) as follows:
```julia
new_colptr = Vector{T}(undef, g.m + 1)
new_rowval = similar(g.rowval) 
Sᵀ = SparseMatrixCSC{T,T}(g.n, g.m, new_colptr, new_rowval, g.rowval)
ftranspose!(Sᵀ, S, identity)
```
This would avoid any non-necessary allocation, but unfortunately it causes segfaults. Presumably because `Sᵀ` and `S` share their vector of `nzval`, which I thought wouldn't be a problem but it is.